### DISCUSSION TITLE
Virtual Displays on Split Regions of a Display

### DISCUSSION THREAD

**Discussion by Iaphetes** [hyprwm/Hyprland#4685] [General]
Hi,

I'm at the moment at getting a large display, which I would want to somewhat permanently divide into multiple "sub-displays".
![image](https://github.com/hyprwm/Hyprland/assets/69191767/92c76570-a492-4772-acb2-49b19b1898b4)
Each application thinks that the area is a display. This means if I fullscreen an application, it only fullscreens to that region of the screen. If it would somehow be possible to on the fly decide if it should to the entire display or the virtual "sub-display" that would be even better.

LeftWM offers this as an options. They call it workspaces (what Hyprland calls workspaces they call tags).
A workspace is by default created covering each display. But you can change that to do an arbitrary configuration.
https://github.com/leftwm/leftwm/wiki/Config#workspaces

Would something like this be possible in Hyprland.

I'd really like to see something like this, as it theoretically allows better use of large displays.

Thanks in advance

**Comment by zakk4223** [comment]
I briefly explored doing something like this with plugins, but there's no way to integrate the built in hyprland layouts (or any third party plugin based layouts)into this system. Hyprland layouts assume they are operating on an entire monitor. 

If I get brave/motivated enough I might try and make modified versions of dwindle and master that work on regions and see how that goes

**Reply by BrokenStandards** [reply]
I would also be interested if you do, as a fellow purchaser of extra wide display thinking it is as good as two when its really more annoying than 1
